src/Sulu/Bundle/PageBundle/Resources/js/containers/Form/fields/SettingsVersions.js

```diff
@@ -40,6 +40,75 @@ class PageSettingsVersions extends React.Component<Props> {
         });
     }
 
+    @computed get resourceKey(): string {
+        const {
+            schemaOptions: {
+                resource_key: {
+                    value: resourceKey,
+                } = {},
+            },
+        } = this.props;
+
+        if (resourceKey === undefined || typeof resourceKey !== 'string') {
+            throw new Error(
+                'The "resource_key" schemaOption is mandatory and must be a string, but received ' +
+                typeof resourceKey + '!'
+            );
+        }
+
+        return resourceKey;
+    }
+
+    @computed get listKey(): string {
+        const {
+            schemaOptions: {
+                list_key: {
+                    value: listKey,
+                } = {},
+            },
+        } = this.props;
+
+        if (listKey === undefined || typeof listKey !== 'string') {
+            throw new Error(
+                'The "list_key" schemaOption is mandatory and must be a string, but received ' +
+                typeof listKey + '!'
+            );
+        }
+
+        return listKey;
+    }
+
+    @computed get userSettingsKey(): string {
+        const {
+            schemaOptions: {
+                user_settings_key: {
+                    value: userSettingsKey,
+                } = {},
+            },
+        } = this.props;
+
+        if (userSettingsKey === undefined || typeof userSettingsKey !== 'string') {
+            throw new Error(
+                'The "user_settings_key" schemaOption is mandatory and must be a string, but received ' +
+                typeof userSettingsKey + '!'
+            );
+        }
+
+        return userSettingsKey;
+    }
+
+    @computed get parentRoute(): string {
+        const {router} = this.props;
+
+        if (!router || !router.route || !router.route.parent || !router.route.parent.name) {
```

Comment:
I think it is safe to use the `?.` operator now, so this could be simplified to `!router?.route?.parent?.name`, if you want to type less :slightly_smiling_face:

Reply:
But when reading this, are you sure this is true for all cases? :thinking: I guess we can leave it like this for now, because this probably works for pages and articles, but maybe we have to introduce a separate parameter for that as well at some point.

Reply:
No it probably doesn't, because not everyone has a parent route...
At first I thought we could also have a `route` as a param in the xml configuration, but this is also not possible because the article bundle uses a different route for each `articleType`.

For now I would also go with this approach.